cgogn/geometry/types/aabb.h

```diff
@@ -358,6 +232,110 @@ std::istream& operator>>(std::istream& in, AABB<VEC_T>& bb)
 	return in;
 }
 
+template <typename VEC_T>
+VEC_T diagonal(const AABB<VEC_T>& bb)
+{
+	cgogn_message_assert(bb.is_initialized(), "Axis-Aligned Bounding box not initialized");
+	return bb.max() - bb.min();
+}
+
+template <typename VEC_T>
+VEC_T center(const AABB<VEC_T>& bb)
+{
+	cgogn_message_assert(bb.is_initialized(), "Axis-Aligned Bounding box not initialized");
+	VEC_T center = (bb.max() + bb.min()) / typename VEC_T::Scalar(2);
+	return center;
+}
+
+// return true if bb intersects the axis-aligned bounding box
+template <typename VEC_T>
+bool intersects(const AABB<VEC_T>& lhs, const AABB<VEC_T>& rhs)
+{
+	cgogn_message_assert(lhs.is_initialized(), "lhs Axis-Aligned Bounding box not initialized");
+	cgogn_message_assert(rhs.is_initialized(), "rhs Axis-Aligned Bounding box not initialized");
+
+	VEC_T lhsmin = lhs.min();
+	VEC_T lhsmax = lhs.max();
+	VEC_T rhsmin = rhs.min();
+	VEC_T rhsmax = rhs.max();
+	for(uint32 i = 0; i < lhs.dim_; ++i)
+	{
+		if(lhsmax[i] < rhsmin[i])
+			return false;
+		if(lhsmin[i] > rhsmax[i])
+			return false;
+	}
+	return true;
+}
+
+// return true if the point belongs strictly to a bounding box
+template <typename VEC_T>
+bool contains(const AABB<VEC_T>& bb, const VEC_T p)
+{
+	cgogn_message_assert(bb.is_initialized(), "Axis-Aligned Bounding box not initialized");
+
+	VEC_T bbmin = bb.min();
+	VEC_T bbmax = bb.max();
```

Comment:
Est-ce qu'on ne devrait pas avoir des getters qui renvoient des const & pour éviter d'avoir à faire des copies dans ces fonctions où on fait de la lecture seule ?

Reply:
min() et max() renvoie au choix une ref ou une const ref. Je peux par contre passer les VEC_T en const si tu veux

Reply:
Mais est-ce qu'il ne faudrait pas écrire `const VEC_T& bbmin = bb.min();` pour ne pas faire de copie locale ?

Reply:
Je viens de lire [ceci (can break encapsulation)](http://thesyntacticsugar.blogspot.com/2011/09/evil-side-of-returning-member-as-const.html) sur le return d'une const ref. Du coup je ne sais pas trop.

Reply:
A part que le `const_cast` c'est le mal, je ne vois pas de problème à retourner une `const &`, à condition de l'affecter à une `const &` si on veut éviter une recopie.